magento/pwa-studio
# [PWA-141] Re-Factor getCartDetails to GraphQL

## Description

Current action is getCartDetails
investigation is necessary to determine full list of used fields from current getCartDetails REST call
currently broken into several requests via `fetchCartPart`
https://devdocs.magento.com/guides/v2.3/graphql/queries/cart.html

## Related Issue





* [[PWA-141](https://jira.corp.magento.com/browse/PWA-141)] Getting Cart Details should be done with graphql Query
## Acceptance 



### Verification Stakeholders

### Specification



### Verification Steps



1. Add Simple Product to Cart
2. Verify that cart indicator reflects item in Cart
3. Edit Quantity and verify change takes
4. Refresh browser and verify that cart doesn't change 
5. Remove item and verify empty cart and indicator reflect change
6. Run through same flow with Configurable Product
7. Add multiple products and verify totals and product counts are correct
8. Open checkout and make sure totals and item count are accurate
9. Perform a checkout, make sure you order goes through (should get an email to confirm)

## Screenshots / Screen Captures (if appropriate)

## Checklist

* I have updated the documentation accordingly, if necessary.
* I have added tests to cover my changes, if necessary.
